src/olympia/versions/models.py

```diff
@@ -253,7 +254,9 @@ def from_upload(cls, upload, addon, selected_apps, channel,
                     'from_api': upload.source == amo.UPLOAD_SOURCE_API,
                 }
             )
-            activity.log_create(amo.LOG.ADD_VERSION, version, addon)
+            activity.log_create(
+                amo.LOG.ADD_VERSION, version, addon,
+                user=upload.user or get_task_user())
```

Comment:
when don't we have `upload.user`?

Reply:
It should always be there, but some tests specifically set it to `None` to force weird situations and see what happens... so I added this fallback cause `ActivityLog` creation without a `user` silently fails...